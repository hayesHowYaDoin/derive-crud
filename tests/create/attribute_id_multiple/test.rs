use derive_crud::Create;

#[derive(Create)]
#[crud_table("test_table")]
struct Test {
    #[crud_id]
    id: i64,
    #[crud_id]
    name: String,
}

#[tokio::main]
async fn main() {
    Test::create("Debbie".to_string()).await;
}
