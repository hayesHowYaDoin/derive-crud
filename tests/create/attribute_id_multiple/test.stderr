error: Only one field can be annotated with `#[crud_id]`
 --> tests/create/attribute_id_multiple/test.rs:8:5
  |
8 | /     #[crud_id]
9 | |     name: String,
  | |________________^

error[E0599]: no function or associated item named `create` found for struct `Test` in the current scope
  --> tests/create/attribute_id_multiple/test.rs:17:19
   |
5  | struct Test {
   | ----------- function or associated item `create` not found for this struct
...
17 |     let _ = Test::create(&pool, "Debbie".to_string()).await;
   |                   ^^^^^^ function or associated item not found in `Test`
